presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationCompiler.java

```diff
@@ -126,6 +129,41 @@ public static BindableAggregationFunction generateAggregationBindableFunction(Cl
         return builder.build();
     }
 
+    private static List<TypeVariableConstraint> parseTypeVariableConstraints(Method inputFunction)
+    {
+        return createTypeVariableConstraints(Arrays.asList(inputFunction.getAnnotationsByType(TypeParameter.class)), parseImplementationDependencies(inputFunction));
+    }
+
+    private static List<ScalarImplementation.ImplementationDependency> parseImplementationDependencies(Method inputFunction)
+    {
+        ImmutableList.Builder<ScalarImplementation.ImplementationDependency> builder = ImmutableList.builder();
+        List<TypeParameter> typeParameters = Arrays.asList(inputFunction.getAnnotationsByType(TypeParameter.class));
+        Set<String> literalParameters = getLiteralParameter(inputFunction);
+
+        for (int i = 0; i < inputFunction.getParameterCount(); i++) {
+            Class<?> parameterType = inputFunction.getParameterTypes()[i];
+            Annotation[] annotations = inputFunction.getParameterAnnotations()[i];
+
+            // Skip injected parameters
+            if (parameterType == ConnectorSession.class) {
+                continue;
+            }
+
+            if (containsMetaParameter(annotations)) {
+                checkArgument(annotations.length == 1, "Meta parameters may only have a single annotation [%s]", inputFunction);
+                Annotation annotation = getOnlyElement(Arrays.asList(annotations));
+                if (annotation instanceof TypeParameter) {
+                    checkArgument(typeParameters.contains(annotation), "Injected type parameters must be declared with @TypeParameter annotation on the method [%s]", inputFunction);
```

Comment:
I've added separate commit to align my code with those recent changes. I hope it is fine to do that as separate commit - that way I can save a lot of time on tracking all the changes. Also the aspect of the change is clearly visible. Let me know if you would like me to spread those changes across commits instead.